src/packaging/tags.py

```diff
@@ -103,13 +103,13 @@ def parse_tag(tag: str) -> FrozenSet[Tag]:
     Returning a set is required due to the possibility that the tag is a
     compressed tag set.
     """
-    tags = set()
     interpreters, abis, platforms = tag.split("-")
-    for interpreter in interpreters.split("."):
-        for abi in abis.split("."):
-            for platform_ in platforms.split("."):
-                tags.add(Tag(interpreter, abi, platform_))
-    return frozenset(tags)
+    return frozenset(
+        Tag(interpreter, abi, platform_)
+        for platform_ in platforms.split(".")
+        for abi in abis.split(".")
+        for interpreter in interpreters.split(".")
+    )
```

Comment:
This form is less clear than the original form, even though it is slightly more efficient. I'd prefer keeping the original form here.

Reply:
I guess it depends on coding habits. I find it less cluttered and easier to read.

Reply:
I'm w/ @pradyunsg ; please keep the original form.